### ISSUE TITLE
Node sharing as service (Non Tailscale users.)

### ISSUE THREAD

**Issue by darshinimashar** [tailscale/tailscale#1676]
The customer would like to use Tailscale to share his machine with users who do not have Tailscale installed as a shareable link. This can make it as easy as possible for the viewer to access the app/service/temporarily...

**Comment by bradfitz** [comment]
I don't think this is something we plan on doing anytime soon.

The Tailscale agent is required to get direct connections between peers through NATs.

Otherwise we'd be proxying everything through the cloud and paying for that bandwidth.

If anything we might make features where users could do this themselves more easily with a $5/mo cloud VM.

/cc @apenwarr @crawshaw

**Comment by apenwarr** [comment]
Yeah, I really like the idea of offering pre-made VMs ("just add an authkey") that you can use to export your services to outsiders. Plus it has the advantage that you never have to add a "tailscale owned" node into your private network. Ideally we would never have to offer a service where our own nodes see a user's unencrypted traffic, even if it's to offer an otherwise valuable service.

Of course, if there's really a lot of customer demand, that might change. As a user-settable option :)

**Comment by DentonGentry** [comment]
At this point, I'm going to close this as I don't expect it to be worked on.

The earlier comment about one-click pre-made VMs to join the Tailnet is much more likely.
